### PR TITLE
fix: cbr for schematics in fscloud

### PR DESCRIPTION
**Pull request by shemau**
### Description

Update fscloud example to extend CBR to include public permissions from schematics

Issue: 6450

### Release required?

- [x] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: examples/fscloud/main.tf [hunk 1]
```diff
@@ -114,6 +132,16 @@ module "cos_fscloud" {
           name  = "networkZoneId"
           value = module.cbr_zone.zone_id
       }]
+      }, {
+      attributes = [
+        {
+          "name" : "endpointType",
+          "value" : "public"
```
Comment (ocofaigh) [inline examples/fscloud/main.tf:139]: not sure we want public here? I believe that would enforce a rule that says schematics IP range can only talk with COS over public endpoint

**Review by ocofaigh — CHANGES_REQUESTED** [review]
we need similar rule for bucket right?

**Comment by ocofaigh** [comment]
OK lets test it out and see. Deploy the fsclud example in schematics manually and ensure the destroy passes. Thanks

**Comment by shemau** [comment]
It looks like the final solution is:

To set the schematics CBR zone and CBR rules to private on both the instance and bucket because the provider is looking up the bucket information. Also to change the bucket management_endpoint_type to private so that terraform uses the private interface on the instance to destroy the bucket.

I have successfully created and destroyed an fscloud example where existing hpcs and activity tracker were used.

**Review by ocofaigh — CHANGES_REQUESTED** [review]
Nice work @shemau 
Can you please update the readme for the fscloud example now that we have added this logic

File: examples/fscloud/variables.tf [hunk 2]
```diff
@@ -50,3 +50,13 @@ variable "access_tags" {
   description = "Optional list of access tags to be added to the created resources"
   default     = []
 }
+
+variable "management_endpoint_type_for_bucket" {
+  description = "The type of endpoint for the IBM terraform provider to use to manage the bucket. (public, private or direct)"
+  type        = string
+  default     = "private"
```
Comment (ocofaigh) [inline examples/fscloud/variables.tf:57]: Wondering if this should default to `public` so would work locally by default? And in [catalogValidationValues.json.template ](https://github.com/terraform-ibm-modules/terraform-ibm-cos/blob/main/examples/fscloud/catalogValidationValues.json.template) we can set it to private as we know that will always run in schematics?

File: tests/pr_test.go [hunk 3]
```diff
@@ -100,6 +100,7 @@ func TestRunFSCloudExample(t *testing.T) {
 	options := setupOptions(t, "cos-fscloud", fsCloudTerraformDir)
 	options.TerraformVars["bucket_existing_hpcs_instance_guid"] = permanentResources["hpcs_south"]
 	options.TerraformVars["bucket_hpcs_key_crn"] = permanentResources["hpcs_south_root_key_crn"]
+	options.TerraformVars["management_endpoint_type_for_bucket"] = "public"
```
Comment (ocofaigh) [inline tests/pr_test.go:103]: as per https://github.com/terraform-ibm-modules/terraform-ibm-cos/pull/480/files#r1433917291 I think we default the variable to public

**Comment by ocofaigh** [comment]
FYI, a release is not needed when updating a test or example (see https://terraform-ibm-modules.github.io/documentation/#/versioning?id=release-versioning)

**Comment by ocofaigh** [comment]
@shemau It seems the way you are writing the `/run pipeline` command, it is not triggering the pipeline. Are you using the saved reply option by any chance? I think there are known issue with that

File: examples/fscloud/variables.tf (hunk 2 above)
Comment (shemau) [reply examples/fscloud/variables.tf:57]: I figured we should default to secure in fscloud and override for our tests. I was thinking, if the CBR is already only from inside VPC, then how does access from github actions work over public? So I am doing some extra checking.

**Comment by shemau** [comment]
> @shemau It seems the way you are writing the `/run pipeline` command, it is not triggering the pipeline. Are you using the saved reply option by any chance? I think there are known issue with that

No, I manually type it and hit the green comment button.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**